### ISSUE TITLE
[jwt-play] `exp` is only applied when `defaultClaim` is used

### ISSUE THREAD

**Issue by mattseburn** [jwt-scala/jwt-scala#165]
I hope I'm missing something obvious here, because it's not clear to me how this would work.

I've set a `maxAge` value in my config:

`play.http.session.maxAge=10s`

However, the `exp` claim doesn't appear when I inspect my jwt.  Here is how I was constructing the `JwtSession`:

```
  def login() = Action(parse.json).async { implicit request: Request[JsValue] =>
    request.body.validate[LoginJsonRequest].fold(
      errors => Future.successful(BadRequest(Json.obj("errors" -> JsError.toJson(errors)))),
      loginJsonRequest => authenticationService.authenticate(loginJsonRequest.username, loginJsonRequest.password)
        .map(_
          .map(user => Ok.withJwtSession(Json.obj("user" -> Json.toJson(UserJsonResponse(user)))))
          .getOrElse(Unauthorized)))
```

Which gives me this jwt:

```
{
  "user": {
    "id": 1,
    "username": "matt"
  }
}
```

However, when I explicitly pass in the `defaultClaim` as a starting point, like this:

```
  def login() = Action(parse.json).async { implicit request: Request[JsValue] =>
    request.body.validate[LoginJsonRequest].fold(
      errors => Future.successful(BadRequest(Json.obj("errors" -> JsError.toJson(errors)))),
      loginJsonRequest => authenticationService.authenticate(loginJsonRequest.username, loginJsonRequest.password)
        .map(_
          .map(user => Ok
            .withJwtSession(JwtSession(JwtSession.defaultHeader, JwtSession.defaultClaim) + Json.obj("user" -> Json.toJson(UserJsonResponse(user)))))
          .getOrElse(Unauthorized)))
```

I'm given this jwt, which includes the `exp` claim:

```
{
  "exp": 1592540162,
  "user": {
    "id": 1,
    "username": "matt"
  }
}
```

After inspecting the code, it appears that the only reference to `MAX_AGE` is in `defaultClaim`:

```
  def defaultClaim(implicit conf: Configuration, clock: Clock): JwtClaim = MAX_AGE match {
    case Some(seconds) => JwtClaim().expiresIn(seconds)
    case _ => JwtClaim()
  }
```

Which is only used when the fields passed to `apply` are not only in key/value format, but are also empty:

```
  def apply(fields: (String, JsValueWrapper)*)(implicit conf:Configuration, clock: Clock): JwtSession =
    if (fields.isEmpty) {
      JwtSession.apply(defaultHeader, defaultClaim)
    } else {
      JwtSession.apply(Json.obj(fields: _*))
    }
```

So, what am I missing??

Thanks!

**Comment by pauldijou** [comment]
Nice investigation, thanks. Could be a bug... There should be [a test for expiration](https://github.com/pauldijou/jwt-scala/blob/master/play/src/test/scala/JwtSessionCustomSpec.scala#L117-L125) though.

Unfortunately, I'm traveling right now, will take a look at it in 2-3 weeks. Don't hesitate to ping me again if I forget (pull request is also super welcome ^^)

**Comment by pauldijou** [comment]
Did my best to fix that in 5.0.0
